strawberryfields/apps/graph/resize.py

```diff
@@ -0,0 +1,234 @@
+
+.. currentmodule:: strawberryfields.future.apps.graph.resize
+
+This module provides functionality for resizing of subgraphs. Samples from
+:func:`~strawberryfields.future.apps.sample.quantum_sampler` have a variable number of clicks (
+equivalently, the number of ones for samples with threshold detection). This results in subgraphs of
+a random size in :func:`~strawberryfields.future.apps.graph.sample.dense_subgraph_sampler_gbs`.
+However, some algorithms may want to work with subgraphs of a fixed size, necessitating the
+resizing of sample subgraphs.
+
+Resizing functionality is provided by the :func:`resize_subgraphs` function, allowing the user to
+make use of different methods for resizing. The available methods are specified in
+:func:`resize_subgraphs`.
+
+Summary
+-------
+
+.. autosummary::
+    METHOD_DICT
+    RESIZE_DEFAULTS
+    resize_subgraphs
+    greedy_density
+    greedy_degree
+
+Code details
+------------
+"""
+
+from typing import Iterable, Optional
+import itertools
+import networkx as nx
+
+from . import utils
+
+RESIZE_DEFAULTS = {"method": "greedy-density"}
+"""Dict[str, Any]: Dictionary to specify default parameters of options for resizing
+"""
+
+
+def resize_subgraphs(
+    subgraphs: Iterable,
+    graph: nx.Graph,
+    target: int,
+    resize_options: Optional[dict] = None,
+) -> list:
+    """Resize subgraphs to a given size.
+
+    This function resizes subgraphs to size ``target``. The resizing method can be fully
+    specified by the optional ``resize_options`` argument, which should be a dict containing any
+    of the following:
+
+    .. glossary::
+
+        key: ``"method"``, value: *str* or *callable*
+            Value can be either a string selecting from a range of available methods or a
+            customized callable function. Options include:
+
+            - ``"greedy-density"``: resize subgraphs by iteratively adding/removing nodes that
+              result in the densest subgraph among all candidates (default)
+            - ``"greedy-degree"``: resize subgraphs by iteratively adding/removing nodes that
+              have the highest/lowest degree.
+            - *callable*: accepting ``(subgraphs: list, graph: nx.Graph,
+              target: int, resize_options: dict)`` as arguments and returning a ``list``
+              containing the resized subgraphs as lists of nodes, see :func:`greedy_density` for
+              an example signature
+
+    Args:
+        subgraphs (iterable[list[int]]): an iterable over subgraphs, where each subgraph is given
+            by a list of nodes
+        graph (nx.Graph): the input graph
+        target (int): the target number of nodes (must be larger than 1)
+        resize_options (dict[str, Any]): dictionary specifying options used during resizing;
+            defaults to :const:`RESIZE_DEFAULTS`
+
+    Returns:
+        list[list[int]]: a list of resized subgraphs, where each subgraph is given by a list of
+        nodes
+    """
```

Comment:
Similarly to the previous module, I think it makes more sense to remove the wrapper function, and simply have two top level functions: `greedy_density_resize` and `greedy_degree_resize`.

Reply:
Sounds good!